File: mvvm/JetLiYu/app/src/main/java/pri/weiqiang/jet/liyujapanese/data/dao/WordDao.java
package pri.weiqiang.jet.liyujapanese.data.dao;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import pri.weiqiang.jet.liyujapanese.data.bean.Word;

@Dao
public interface WordDao {

    @Query("SELECT * FROM words WHERE lesson_id LIKE :lessonId")
    LiveData<List<Word>> getWordByLessonId(String lessonId);

    @Query("SELECT * FROM words ")
    LiveData<List<Word>> getAll();

    @Query("SELECT * FROM words WHERE _id LIKE:id ")
    LiveData<Word> getWordById(int id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Void insertAll(Word... words);


}

File: mvvm/KtJetLiYu/app/src/main/java/pri/weiqiang/jet/kt/liyujapanese/data/dao/WordDao.kt
package pri.weiqiang.jet.kt.liyujapanese.data.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import pri.weiqiang.jet.kt.liyujapanese.data.bean.Word

@Dao
interface WordDao {

    @Query("SELECT * FROM words WHERE lesson_id LIKE :lessonId")
    fun getWordByLessonId(lessonId:String):LiveData<List<Word>>

    @Query("SELECT * FROM words ")
    fun getAll():LiveData<List<Word>>

    @Query("SELECT * FROM words WHERE _id LIKE:id ")
    fun getWordById(id:Int):LiveData<List<Word>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(vararg words:Word)

}